src/row.rs

```diff
@@ -4,22 +4,42 @@ use chrono::{DateTime, Days, NaiveDate, NaiveDateTime};
 
 use crate::{Error, Result};
 
+#[derive(Debug)]
+pub struct SnowflakeColumnType {
+    /// The index of the column in the row
+    pub index: usize,
+    /// Data type of the column in Snowflake
+    pub snowflake_type: String,
+    /// Whether the column is nullable
+    pub nullable: bool,
+}
+
 #[derive(Debug)]
 pub struct SnowflakeRow {
     pub(crate) row: Vec<Option<String>>,
-    pub(crate) column_names: Arc<HashMap<String, usize>>,
+    pub(crate) column_types: Arc<HashMap<String, SnowflakeColumnType>>,
 }
 
 impl SnowflakeRow {
     pub fn get<T: SnowflakeDecode>(&self, column_name: &str) -> Result<T> {
-        let index = self
-            .column_names
+        let column_type = self
+            .column_types
             .get(&column_name.to_ascii_uppercase())
             .ok_or_else(|| Error::Decode(format!("column not found: {}", column_name)))?;
-        self.row[*index].try_get()
+        self.row[column_type.index].try_get()
     }
     pub fn column_names(&self) -> Vec<&str> {
-        self.column_names.iter().map(|(k, _)| k.as_str()).collect()
+        self.column_types.iter().map(|(k, _)| k.as_str()).collect()
+    }
+    pub fn column_types(&self) -> Vec<(&str, &SnowflakeColumnType)> {
```

Comment:
I feel like it's not clear for users that the first element of each tuple is the name of the column. Would you mind creating a new struct and putting the type and name into it, like this:

```rust

pub struct SnowflakeColumn {
    name: String,
    column_type: SnowflakeColumnType,
}

pub fn columns(&self) -> Vec<SnowflakeColumn> {
    ...
```

Reply:
Or, we might want to put the `name` into the struct, like this:
```rust
pub struct SnowflakeColumn {
    /// The index of the column in the row
    pub index: usize,
    /// Data type of the column in Snowflake
    pub snowflake_type: String,
    /// Whether the column is nullable
    pub nullable: bool,

   pub name: String,
}
```

Reply:
Fixed to adopt the first one